pkgs/applications/science/logic/lean2/default.nix

```diff
@@ -1,19 +1,26 @@
-{ lib, stdenv, fetchFromGitHub, cmake, gmp, mpfr, python2
-, gperftools, ninja, makeWrapper }:
+{ lib, stdenv, fetchpatch, fetchFromGitHub, cmake, gmp, mpfr, python3
+, jemalloc, ninja, makeWrapper }:
 
 stdenv.mkDerivation {
   pname = "lean2";
-  version = "2017-07-22";
+  version = "2018-10-01";
 
   src = fetchFromGitHub {
     owner  = "leanprover";
     repo   = "lean2";
-    rev    = "34dbd6c3ae612186b8f0f80d12fbf5ae7a059ec9";
-    sha256 = "1xv3j487zhh1zf2b4v19xzw63s2sgjhg8d62a0kxxyknfmdf3khl";
+    rev    = "8072fdf9a0b31abb9d43ab894d7a858639e20ed7";
+    sha256 = "12bscgihdgvaq5xi0hqf5r4w386zxm3nkx1n150lv5smhg8ga3gg";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "lean2-fix-compilation-error.patch";
+      url = "https://github.com/leanprover/lean2/pull/13.patch";
```

Comment:
```suggestion
      url = "https://github.com/leanprover/lean2/commit/09b316ce75fd330b3b140d138bcdae2b0e909234.patch";
```

Reply:
Given that this has not yet been merged, I think pointing to `leanprover/lean2` instead of `collares/lean2` is misleading. I know that this is how GitHub does forks, but still. Would you be OK with `https://github.com/collares/lean2/commit/09b316ce75fd330b3b140d138bcdae2b0e909234.patch` instead?

Reply:
works for me as long as you keep your fork.

Reply:
Done. I was waiting for upstream to merge the PR, just in case, but it didn't pan out.